datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxController.java

```diff
@@ -336,7 +338,29 @@ public SdxBackupStatusResponse backupDatalakeStatusByName(@ResourceName String n
     public SdxDatabaseRestoreResponse restoreDatabaseByName(@ResourceName String name, String backupId,
             String restoreId, String backupLocation) {
         SdxCluster sdxCluster = getSdxClusterByName(name);
-        return sdxBackupRestoreService.triggerDatabaseRestore(sdxCluster, backupId, restoreId, backupLocation);
+        try {
+            SdxDatabaseRestoreStatusResponse response = sdxBackupRestoreService.getDatabaseRestoreStatus(sdxCluster, restoreId);
+            SdxDatabaseRestoreResponse sdxDatabaseRestoreResponse = new SdxDatabaseRestoreResponse();
+            sdxDatabaseRestoreResponse.setOperationId(restoreId);
+            return sdxDatabaseRestoreResponse;
+        } catch (NotFoundException notFoundException) {
+            return sdxBackupRestoreService.triggerDatabaseRestore(sdxCluster, backupId, restoreId, backupLocation);
+        }
+    }
+
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.RESTORE_DATALAKE)
+    public SdxRestoreResponse restoreDatalakeByName(@ResourceName String name, String backupId, String backupLocation, String backupLocationOverride) {
+        SdxCluster sdxCluster = getSdxClusterByName(name);
+        return sdxBackupRestoreService.triggerDatalakeRestore(sdxCluster, backupId, backupLocation, backupLocationOverride);
+    }
+
+    @Override
+    @CheckPermissionByResourceName(action = AuthorizationResourceAction.RESTORE_DATALAKE)
+    public SdxRestoreStatusResponse restoreDatalakeStatusByName(@ResourceName String name,
```

Comment:
pls rename to getRestoreDatalakeStatusByName

Reply:
Thanks. Corrected!